doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,207 @@
+
+To have a better user experience, vic-machine should notify user for what's changed, for example, list of volumes will be added, and another list will be removed.
+
+Pros of this option is no new options introduced and all options used in vic-machine create could be reused in update command.
+
+Cons of this option is that every time to update a VCH, user need to know all the old configuration first, otherwise some configuration will be removed by mistake.
+
+
+If there are many options specified to create VCH, and only a few things need to be tuned, the delta update option is more clear, and less effort to write the update command. But if there are less options specified in first create, and lots of changes needed, the whole update option will be easier to know what will be there after update.
+
+### Separate Update to Sub Commands
+The idea of this option is to update a few configuration each time, which is similar to Delta Options, but we're not using different options to separate the update scope, instead we can use different update sub command. For example, to update volume-store, we can use following command
+```
+vic-machine update volume-store --rm ds://datastore/volume:default
+vic-machine update volume-store --add nfs://host:port/container:nfs
+```
+
+The command format is like ```vic-machine update <update object> --<update keyword> --<update option> <update value>```
+
+Pros:
+- The whole update options are splitted to many update sub commands. Then in each sub command, only a small number of options are available, and that does not increase over time. Only the sub command number will increase.
+- Consistent with existing update firewall command format
+
+Cons:
+- We'll need to reinvent option names
+
+  For example, to update client network, user need to specify all options if they want to use static ip. Then in one update command, we'll need all those options. Here is the command looks like ```vic-machine update bridge-network --port-group value --gateway value --ip value```
+  The existing option name in vic-machine create is ```--client-network value   --client-network-gateway value  --client-network-ip value```, we'll need to remove the prefix cause that is already shown as sub-command name.
+
+Preference:
+
+- The sub commands option
+
+## Show Existing VCH configuration
+Right now, there is no user friendly way to view all configurations of existing VCH. To support update operation, we need to make it easier, otherwise, update only makes things weird.
+
+vic-machine inspect command is available now, just it shows only the execution endpoint of VCH, instead of the configuration. We can extend that command, to support configuration inspection as well.
+
+- vic-machine inspect <--configuration, --conf>
+  show VCH configuration only
+- vic-machine inspect <--execution, -e>
+  show VCH execution endpoint (default behavior)
+
+### Configuration Inspection
+For configuration inspection, two output format should be supported
```

Comment:
This is basically generating the vic-machine manifest that we spoke about last year - if we do this we should take care to structure it in such a way that it can be used as such:
* input for creating a new VCH
* the format doesn't prohibit use for authority delegation.